Sources/SwiftInspectorCore/PropertyAnalyzer.swift

```diff
@@ -0,0 +1,248 @@
+    return .visitChildren
+  }
+
+  override func visit(_ node: StructDeclSyntax) -> SyntaxVisitorContinueKind {
+    processNode(node, withName: node.identifier.text, members: node.members.members)
+    return .visitChildren
+  }
+
+  override func visit(_ node: ExtensionDeclSyntax) -> SyntaxVisitorContinueKind {
+    if let typeIdentifier = node.extendedType.as(SimpleTypeIdentifierSyntax.self) {
+      processNode(node, withName: typeIdentifier.name.text, members: node.members.members)
+    }
+    return .visitChildren
+  }
+
+  // MARK: Private
+
+  private let onNodeVisit: (_ info: TypeWithPropInfo) -> Void
+
+  private func processNode<Node>(_ node: Node, withName name: String, members: MemberDeclListSyntax) where Node: SyntaxProtocol {
+    var result = [TypeWithPropInfo.PropInfo]()
+    let propertyVisitor = PropertySyntaxVisitor(parentName: name) { info in
+      result.append(info)
+    }
+    propertyVisitor.walk(node)
+    onNodeVisit(.init(name: name, properties: result))
+  }
+}
+
+// MARK: - PropertySyntaxVisitor
+
+private final class PropertySyntaxVisitor: SyntaxVisitor {
+
+  init(parentName: String, onNodeVisit: @escaping (_ info: TypeWithPropInfo.PropInfo) -> Void) {
+    self.parentName = parentName
+    self.onNodeVisit = onNodeVisit
+  }
+
+  override func visit(_ node: VariableDeclSyntax) -> SyntaxVisitorContinueKind {
+    var leadingTrivia: Trivia? = node.leadingTrivia
+    var access: TypeWithPropInfo.Access = .internal
+    var scope: TypeWithPropInfo.Scope = .instance
+
+    node.modifiers?.forEach { modifier in
+      // leading trivia is on the first modifier, or the node itself if no modifiers are present
+      leadingTrivia = leadingTrivia ?? modifier.leadingTrivia
+      // This attempts to get any access information from the node's modifiers
+      if let modifierAccess = TypeWithPropInfo.Access(rawValue: modifier.name.text) {
+        access = modifierAccess
+      }
+      // this attempts to get any scope information from the node's modifiers
+      if let modifierScope = TypeWithPropInfo.Scope(rawValue: modifier.name.text) {
+        scope = modifierScope
+      }
+    }
+
+    // in practice this `bindings` array only has 1 element
+    node.bindings.forEach { binding in
+      if let identifier = binding.pattern.as(IdentifierPatternSyntax.self) {
+        if
+          let typeAnnotation = binding.typeAnnotation,
+          let simpleTypeIdentifier = typeAnnotation.type.as(SimpleTypeIdentifierSyntax.self)
+        {
+          onNodeVisit(.init(
+            name: identifier.identifier.text,
+            typeAnnotation: simpleTypeIdentifier.name.text,
+            comment: comment(from: leadingTrivia),
+            access: access,
+            scope: scope))
+          return
+        }
+        // no type annotation was found. In this case it's much more complicated to get
+        // the type information for a variable.
+        // TODO: Include logic to get types of any variable declaration
```

Comment:
What's an example of this use case?

Reply:
I can include this in the comments, but this is an example where we can easily find the type:
```swift
public static let thing: String = "Hello"
```
whereas this doesn't have the type information:
```swift
public static let thing = "Hello"
```
I don't really know how to handle the latter case here, so I've just chosen to pass over it for the time being. What are your thoughts?

Reply:
I think it makes sense to not pass the type here, This makes sense! Let's add this as a comment 🙏